src/main/java/org/reactome/addlinks/fileprocessors/pharos/PharosLigandsFileProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.reactome.addlinks.fileprocessors.pharos;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.reactome.addlinks.fileprocessors.FileProcessor;
+
+public class PharosLigandsFileProcessor extends FileProcessor<String>
+{
+	public PharosLigandsFileProcessor(String processorName)
+	{
+		super(processorName);
+	}
+
+	public PharosLigandsFileProcessor()
+	{
+		super(null);
+	}
+
+	@Override
+	public Map<String, String> getIdMappingsFromFile()
+	{
+		Map<String, String> mapping = new HashMap<>();
+
+		try(BufferedReader reader = new BufferedReader(new FileReader(this.pathToFile.toFile())))
+		{
+			String line = reader.readLine();
+			while (line != null)
+			{
+				String[] parts = line.split("\t");
+				// first value is Guide to Pharmacology identifier, second value is ligid (Pharos ligand identifier).
+				mapping.put(parts[0], parts[1]);
+				line = reader.readLine();
+			}
+		}
+		catch (FileNotFoundException e)
+		{
+			logger.error("File {} could not be found, file processor has failed.", this.pathToFile.toString());
+		}
+		catch (IOException e)
+		{
+			logger.error("Error with file! File processing has probably failed.", e);
+		}
```

Comment:
For consistency, can the exception object `e` be logged on line 42?

Reply:
I didn't like the Exception object because I felt that a `FileNotFoundException`, combined with the name of the file on the path that it was _supposed_ to be found on was already pretty clear. I logged the `IOException` object because there could be more than one line of code that triggers it, and there could be more than one cause for it, so I felt that logging the exception object was warranted in this case.

Reply:
Okay, would the logged error indicate what class triggered it though so it's clear where in code the file that wasn't findable is being used?

Reply:
Passing the exception to logger.error should include the exception's stack trace in the log message.

Reply:
Okay, sounds good.